### DISCUSSION TITLE
Undocumented windows prerequisite: Build 21H2

### DISCUSSION THREAD

**Discussion by Gamekiller48** [moyix/fauxpilot#116] [General]
Hi, I've been trying to get this project to run on my hardware, unfortunately however I seem to be running into an issue whenever I try:
Launching the project (via launch.sh or via docker desktop) results in the following error message:

(error seems to originate from the triton-1 container):
Error invoking remote method 'docker-start-container': Error: (HTTP code 500) server error - failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error running hook #0: error running hook: exit status 1, stdout: , stderr: Auto-detected mode as 'legacy' nvidia-container-cli: initialization error: WSL environment detected but no adapters were found: unknown

System:
Windows 10 Pro x64 Build 21H1 
CPU:  i7-12700K
GPU: GTX 1080

I think I managed to indentify the issue via [this thread](https://github.com/NVIDIA/nvidia-docker/issues/1692#issuecomment-1298322661), which indicates that there is some issue with WSL2 that prevents it from accessing CUDA on a GTX 1080 without the 21H2 update installed.  
Unfortunately for me, installing this update doesn't work on my machine (it straight up doesn't show up as an available update under the updates section), which means that I sadly won't be able to test out this project. Thought I might document my problems here anyways for any other unfortunate soul that might run into the same issue.

**Comment by phken91** [comment]
I have just set up Fauxpilot for testing on WSL (Ubuntu). These were the steps I followed to complete the test:

1. Installed Docker with docker compose plugin.
2. Followed the steps on this website: https://ubuntu.com/tutorials/enabling-gpu-acceleration-on-ubuntu-on-wsl2-with-the-nvidia-cuda-platform#1-overview.
3. Updated the NVIDIA driver on my host machine (which is Windows 11) to the latest version.
Installed the `nvidia-container-toolkit`.
4. Cloned the Fauxpilot project from Git and ran `setup.sh`.
5. Ran the command `docker compose pull` to download a large file of around 4GB.
Ran `launch.sh`.
